LocalPackages/SyncUI/Sources/SyncUI/Views/Internal/QRCodeScannerView.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  QRCodeScannerView.swift
+//  DuckDuckGo
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import SwiftUI
+import AVFoundation
+
+struct QRCodeScannerView: UIViewRepresentable {
+
+    var onQRCodeScanned: (String) -> Bool
+    var onCameraUnavailable: () -> Void
+
+    func makeCoordinator() -> Coordinator {
+        return Coordinator(self)
+    }
+
+    func makeUIView(context: Context) -> UIView {
+        let view = AutoResizeLayersView()
+        context.coordinator.start(view)
+        return view
+    }
+
+    func updateUIView(_ uiView: UIView, context: Context) {
+    }
+
+    static func dismantleUIView(_ uiView: UIView, coordinator: Coordinator) {
+        coordinator.stop()
+    }
+
+    class Coordinator: NSObject, AVCaptureMetadataOutputObjectsDelegate {
+
+        let session: AVCaptureSession
+        let metadataOutput = AVCaptureMetadataOutput()
+        let cameraView: QRCodeScannerView
+
+        init(_ cameraView: QRCodeScannerView) {
+            self.cameraView = cameraView
+            self.session = AVCaptureSession()
+            super.init()
+        }
+
+        func start(_ uiView: UIView) {
+            session.sessionPreset = .high
+
+            guard let backCamera = AVCaptureDevice.default(.builtInWideAngleCamera, for: .video, position: .back),
+                  let input = try? AVCaptureDeviceInput(device: backCamera) else {
```

Comment:
The camera view doesn’t honor device orientation – I’m testing on an iPad in landscape and am seeing a rotated preview. Feel free to ignore if you’re going to address landscape mode in a separate PR.

![IMG_B8C3F4FE6DD0-1](https://user-images.githubusercontent.com/14022/222404391-fb1365f8-cecb-485f-b5cd-2a306e8e88c0.jpeg)